diff --git a/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_demo_front/dkan_sitewide_demo_front.module b/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_demo_front/dkan_sitewide_demo_front.module
index e6b191f..cbb425a 100644
--- a/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_demo_front/dkan_sitewide_demo_front.module
+++ b/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_demo_front/dkan_sitewide_demo_front.module
@@ -96,6 +96,10 @@ function dkan_sitewide_demo_front_form_alter(&$form, &$form_state, $form_id) {
     $form['#info']['filter-search_api_views_fulltext']['description'] = t('Search');
     $form['#info']['filter-search_api_views_fulltext']['label'] = '';
     $form['submit']['#title'] = '';
+    if (isset($form_state['pager_plugin'])) {
+      $form['submit']['#type'] = 'image_button';
+      $form['submit']['#src'] = isset($form_state['pager_plugin']) ? 'profiles/dkan/images/icon-search-27x26.png' : '';
+    }
   }
 }
 
diff --git a/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_menu/dkan_sitewide_menu.info b/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_menu/dkan_sitewide_menu.info
index 7700651..5c407da 100644
--- a/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_menu/dkan_sitewide_menu.info
+++ b/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_menu/dkan_sitewide_menu.info
@@ -3,6 +3,7 @@ core = 7.x
 package = DKAN Features
 description = Default menu settings for DKAN
 dependencies[] = dkan_dataset
+dependencies[] = dkan_dataset_groups
 dependencies[] = dkan_datastore
 dependencies[] = dkan_sitewide_search_db
 dependencies[] = features
