diff --git a/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.blocks.inc b/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.blocks.inc
index f68d03c..e792120 100644
--- a/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.blocks.inc
+++ b/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.blocks.inc
@@ -70,7 +70,7 @@ function dkan_sitewide_block_view($delta = '') {
       break;
 
     case 'dkan_sitewide_user_menu':
-      $block['subject'] = t('');
+      $block['subject'] = '';
       $block['content'] = dkan_sitewide_user_menu_block();
       break;
 
@@ -80,12 +80,17 @@ function dkan_sitewide_block_view($delta = '') {
 
 /**
  * Creates "Additional" block for resources.
+ *
  * TODO: Make this a faux field as well attached to resource node.
  */
-function dkan_sitewide_resource_additional_block($nid  = '') {
+function dkan_sitewide_resource_additional_block($nid = '') {
   $node = $nid ? node_load($nid) : menu_get_object();
   $file_info = array();
-  if (isset($node->field_upload[DKAN_FIELD_LANG]) && $file = $node->field_upload[DKAN_FIELD_LANG][0]) {
+  $lang = $node->language ? $node->language : 'und';
+  $link = isset($node->field_link_remote_file) ? $node->field_link_remote_file[$lang][0] : NULL;
+  $upload = $node->field_upload ? $node->field_upload[$lang][0] : NULL;
+
+  if (($file = $link) || ($file = $upload)) {
     $file_info[] = array(t('mimetype'), $file['filemime']);
     $file_info[] = array(t('filesize'), $file['filesize']);
     $file_info[] = array(t('resource type'), 'file upload');
@@ -105,7 +110,7 @@ function dkan_sitewide_license_block() {
   }
   else {
     if (isset($node->field_license) && $node->field_license) {
-      $key = $node->field_license[DKAN_FIELD_LANG][0];
+      $key = $node->field_license[$node->language][0];
       $field = field_info_instance('node', 'field_license', 'dataset');
       $raw_options = $field['widget']['settings']['available_options'];
       foreach (explode("\n", $raw_options) as $option) {
diff --git a/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.module b/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.module
index d6e71e6..770dd50 100644
--- a/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.module
+++ b/profiles/dkan/modules/dkan/dkan_sitewide/dkan_sitewide.module
@@ -18,16 +18,26 @@ function dkan_sitewide_user_menu_block() {
     if (module_exists('gravatar')) {
       $output = l('<img alt="my user picture" src="' . gravatar_get_gravatar($user->mail, array('size' => '25')) . '">' , 'user', array('html' => TRUE));
     }
-    $output.= '<span class="links">' . l($user->name, 'user') . l('Logout', 'user/logout', array('attributes' => array('class' => array('sub')))) . '</span>';
+    $output .= '<span class="links">' . l($user->name, 'user') . l('Logout', 'user/logout', array('attributes' => array('class' => array('sub')))) . '</span>';
   }
   else {
-    $output.= '<span class="links">' . l('Login', 'user/login') . l('Register', 'user/register', array('attributes' => array('class' => array('sub')))) . '</span>';
+    $output .= '<span class="links">' . l('Login', 'user/login') . l('Register', 'user/register', array('attributes' => array('class' => array('sub')))) . '</span>';
   }
   return $output;
 }
 
 /**
- * Implements hook_page_alter();
+ * Implements hook_cron().
+ */
+function dkan_sitewide_cron() {
+  if (module_exists('defaultcontent')) {
+    module_disable(array('dkan_default_content', 'defaultcontent'));
+    drupal_uninstall_modules(array('dkan_default_content', 'defaultcontent'));
+  }
+}
+
+/**
+ * Implements hook_page_alter().
  */
 function dkan_sitewide_page_alter(&$page) {
   // Adding page title to breadcrumb.
diff --git a/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_demo_front/dkan_sitewide_demo_front.module b/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_demo_front/dkan_sitewide_demo_front.module
index e6b191f..cbb425a 100644
--- a/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_demo_front/dkan_sitewide_demo_front.module
+++ b/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_demo_front/dkan_sitewide_demo_front.module
@@ -96,6 +96,10 @@ function dkan_sitewide_demo_front_form_alter(&$form, &$form_state, $form_id) {
     $form['#info']['filter-search_api_views_fulltext']['description'] = t('Search');
     $form['#info']['filter-search_api_views_fulltext']['label'] = '';
     $form['submit']['#title'] = '';
+    if (isset($form_state['pager_plugin'])) {
+      $form['submit']['#type'] = 'image_button';
+      $form['submit']['#src'] = isset($form_state['pager_plugin']) ? 'profiles/dkan/images/icon-search-27x26.png' : '';
+    }
   }
 }
 
diff --git a/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_menu/dkan_sitewide_menu.info b/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_menu/dkan_sitewide_menu.info
index 7700651..5c407da 100644
--- a/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_menu/dkan_sitewide_menu.info
+++ b/profiles/dkan/modules/dkan/dkan_sitewide/modules/dkan_sitewide_menu/dkan_sitewide_menu.info
@@ -3,6 +3,7 @@ core = 7.x
 package = DKAN Features
 description = Default menu settings for DKAN
 dependencies[] = dkan_dataset
+dependencies[] = dkan_dataset_groups
 dependencies[] = dkan_datastore
 dependencies[] = dkan_sitewide_search_db
 dependencies[] = features
